runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZoneOverlay.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2018, Nickolaj <https://github.com/fire-proof>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.nightmarezone;
+
+import net.runelite.api.Client;
+import net.runelite.api.ItemID;
+import net.runelite.api.Varbits;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.BackgroundComponent;
+import net.runelite.client.ui.overlay.components.TextComponent;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.awt.*;
+
+class NightmareZoneOverlay extends Overlay
+{
+	private static final int WIDTH = 140;
+	private static final int TOP_BORDER = 2;
+	private static final int BOTTOM_BORDER = 3;
+
+	private final Client client;
+	private final NightmareZoneConfig config;
+	private final NightmareZonePlugin plugin;
+	private final InfoBoxManager infoBoxManager;
+	private final ItemManager itemManager;
+
+	private AbsorptionCounter absorptionCounter;
+
+	@Inject
+	NightmareZoneOverlay(
+			@Nullable Client client,
```

Comment:
No need for having client nullable here

Reply:
Don't most of the existing plugins mark `Client` as `@Nullable` when it's being used in their overlays' injected constructor?

Reply:
Yeah, well, but it isn't required. The @Nullable was required before because we were trying to load plugins even when outdated, but not anymore, so the client is always non-null.